crates/js-sys/src/lib.rs

```diff
@@ -1500,6 +1526,35 @@ extern "C" {
     pub fn value_of(this: &Number) -> f64;
 }
 
+macro_rules! number_from {
+    ($($x:ident)*) => ($(
+        impl From<$x> for Number {
+            fn from(x: $x) -> Number {
+                Number { obj: x.into() }
+            }
+        }
+
+        impl PartialEq<$x> for Number {
+            fn eq(&self, other: &$x) -> bool {
+                self.value_of() == f64::from(*other)
+            }
+        }
+    )*)
+}
+number_from!(i8 u8 i16 u16 i32 u32 f32 f64);
```

Comment:
Perhaps a silly suggestion, but why not use transitive generics instead of a custom macro? Something like:

```rust
impl<T> From<T> for Number where f64: From<T> {
  // ...
}
```

The only downside it has is that it makes it impossible to implement custom `From<SomeType> for Number` but I guess we don't really want to support source types that are not convertible to `f64` too?

Reply:
That's possible yeah, but I think I'd prefer to stick with the macro strategy to avoid blanket impls unnecessarily blocking out other impls.